assets/app/scripts/controllers/overview.js

```diff
@@ -384,7 +416,114 @@ angular.module('openshiftConsole')
       });
     });
 
+    /*
+     * The topology graph wants unique identifiers.
+     *
+     * One of the warts on the kubernetes/openshift API is that there
+     * is no unique global identifier per resource. We construct them here from
+     * Kind/metadata.name ... which is good enough in a single namespace.
+     */
+    function makeId(name, kind) {
```

Comment:
The problem is that even if in theory a Deployment *is* a ReplicationController ... you still can't use a ```metadata.uid``` to refer to loaded resource objects (ie: loaded items) uniquely.

Even if you then assume that you will never use a ReplicationController and a Deployment resource objects together in a scenario where you would need unique identifiers ... you run straight into the second problem:

There are objects in kubernetes that do not have metadata.uid set. For example the 'kubernetes' and 'kubernetes-ro' services in the 'default' namespace.